api/src/handlers/utils.rs

```diff
@@ -79,63 +82,27 @@ pub fn get_output_v2(
 	id: &str,
 	include_proof: bool,
 	include_merkle_proof: bool,
-) -> Result<(OutputPrintable, OutputIdentifier), Error> {
-	let c = util::from_hex(String::from(id)).context(ErrorKind::Argument(format!(
-		"Not a valid commitment: {}",
-		id
-	)))?;
-	let commit = Commitment::from_vec(c);
+) -> Result<Option<(OutputPrintable, OutputIdentifier)>, Error> {
+	let chain = w(chain)?;
+	let (_, output_pos, identifier) = match retrieve_output(&chain, id)? {
+		Some(x) => x,
+		None => return Ok(None),
+	};
 
-	// We need the features here to be able to generate the necessary hash
-	// to compare against the hash in the output MMR.
-	// For now we can just try both (but this probably needs to be part of the api
-	// params)
-	let outputs = [
-		OutputIdentifier::new(OutputFeatures::Plain, &commit),
-		OutputIdentifier::new(OutputFeatures::Coinbase, &commit),
-	];
```

Comment:
This always felt kind of awkward. Are we still doing two lookups to see if either match? Or is this not necessary?

Reply:
It's still there, just moved to a different function (`retrieve_output()`). I agree its awkward, we can get rid of it by changing the `is_unspent()` on txhashset to only take a commitment and ignore the output features from the output mmr

Reply:
I think that might actually make sense (to ignore the features).
Let me think about this a bit more. There is a weird edge case with this where we do not allow duplicate outputs, but we can spend one and replace it with another, with the new one having different features. At that point is the "output" spent or not?
From the wallet perspective they are different, but from the node impl we spend based purely on commitment.

Reply:
We should leave this as-is for now - its a material change to ignore the features here (and may have other implications). Either way it should be a separate change to this PR.